src/Relation/Binary/Rewriting.agda

```diff
@@ -0,0 +1,153 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Concepts from rewriting theory
+-- Definitions are based on "Term Rewriting Systems" by J.W. Klop
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.Rewriting where
+
+open import Agda.Builtin.Equality using (_≡_ ; refl)
+open import Data.Product using (_×_ ; ∃ ; -,_; _,_ ; proj₁ ; proj₂)
+open import Data.Empty
+open import Data.Sum as Sum using (_⊎_; inj₁; inj₂)
+open import Function using (flip; _∘′_)
+open import Induction.WellFounded
+open import Level
+open import Relation.Binary.Core
+open import Relation.Binary.Construct.Closure.Reflexive
+open import Relation.Binary.Construct.Closure.ReflexiveTransitive as RTrans
+open import Relation.Binary.Construct.Closure.Symmetric
+open import Relation.Binary.Construct.Closure.Transitive
+open import Relation.Nullary
+
+-- The following definitions are taken from Klop [5]
+module _ {a ℓ} {A : Set a} (_⟶_ : Rel A ℓ) where
+
+  private
+    _⟵_  = flip _⟶_
+    _—↠_ = Star _⟶_
+    _↔_  = Star (SymClosure _⟶_)
```

Comment:
Is this not just `EqClosure` from `Relation.Binary.Construct.Closure.Equivalence`?

Reply:
Didn't realize that existed, using that instead!